modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -240,8 +237,11 @@ public class PartitionReplicaListener implements ReplicaListener {
 
     private final CatalogService catalogService;
 
-    /** Listener for configuration indexes, {@code null} if the replica is not the leader. */
-    private final AtomicReference<ConfigurationNamedListListener<TableIndexView>> indexesConfigurationListener = new AtomicReference<>();
+    /** Listener for creating an index in catalog, {@code null} if the replica is not the leader. */
+    private final AtomicReference<EventListener<CreateIndexEventParameters>> createIndexListener = new AtomicReference<>();
```

Comment:
Do we need to use `AtomicReferenceFieldUpdater` here? `PartitionReplicaListener` is created for every partition

Reply:
`AtomicReferenceFieldUpdater` is an optimization, and at this stage of development we do not save on memory.
If we do optimization, then it is worth doing it under a separate ticket.

Reply:
Sure, I just thought that this optimization looks useful here and does not take a lot of code to implement.